MediaGalleryUi/view/adminhtml/web/js/grid/columns/image/actions.js

```diff
@@ -70,7 +71,58 @@ define([
          * @param {Object} record
          */
         deleteImageAction: function (record) {
-            deleteImage.deleteImageAction(record, this.imageModel().deleteImageUrl);
+            var imageDetailsUrl = this.imageModel().imageDetailsurl,
+                deleteImageUrl = this.imageModel().deleteImageUrl,
+                defaultConfirmationContent = $.mage.__('Are you sure you want to delete "' + record.path + '" image?');
+
+            getDetails(imageDetailsUrl, record.id).then(function(imageDetails) {
+                var confirmationContent = this.getConfirmationContentByImageDetails(imageDetails)
+                    .concat(defaultConfirmationContent);
+                deleteImage.deleteImageAction(record, deleteImageUrl, confirmationContent);
+            }.bind(this)).fail(function () {
+                deleteImage.deleteImageAction(record, deleteImageUrl, defaultConfirmationContent);
+            });
+        },
+
+        /**
+         * Returns confirmation content with information about related content
+         *
+         * @param {Object} imageDetails
+         * @return String
+         */
+        getConfirmationContentByImageDetails: function (imageDetails) {
+            var details = imageDetails.details;
+
+            if(_.isObject(details) && !_.isUndefined(details['6'])) {
+                var detailValue = details['6'].value;
+
+                return $.mage.__(this.getRecordRelatedContentHtml(detailValue));
+            }
+
+            return '';
+        },
+
+        /**
+         * Get information about image use
+         *
+         * @param {Object|String} value
+         * @return {String}
+         */
+        getRecordRelatedContentHtml: function (value) {
+            var usedIn = 'This image is used in ';
+
+            if (_.isObject(value) && !_.isEmpty(value)) {
+                _.each(value, function (numberOfTimeUsed, moduleName) {
+                    usedIn += numberOfTimeUsed + ' ' + moduleName + ' ';
+                });
```

Comment:
```suggestion
        getRecordRelatedContentMessage: function (value) {
            var usedInMessage = $.mage.__('This image is used in %s.'),
                usedIn = '';

            if (_.isObject(value) && !_.isEmpty(value)) {
                _.each(value, function (numberOfTimeUsed, moduleName) {
                    usedIn += numberOfTimeUsed + ' ' + moduleName + ', ';
                });
                usedIn = usedIn.replace(/,\s*$/, "");
                
                return  usedInMessage.replace('%s', usedIn);
            }

            return '';
        }
```

Some small suggestion here as well:
1. Changed the usedIn message to be translatable.
2. Added a comma to each added string and removed the last comma when the message is ready
3. Removed the else and just return empty string instead of message.